main.go

```diff
@@ -277,8 +283,72 @@ func (p *prometheus) Collect(ch chan<- registry.Metric) {
 	}
 }
 
+func usage() {
+	groups := make(map[string][]*flag.Flag)
+	// Set a default group for ungrouped flags.
+	groups["."] = make([]*flag.Flag, 0)
+
+	// Bucket flags into groups based on the first of their dot-separated levels.
+	flag.VisitAll(func(fl *flag.Flag) {
+		parts := strings.SplitN(fl.Name, ".", 2)
+		if len(parts) == 1 {
+			groups["."] = append(groups["."], fl)
+		} else {
+			name := parts[0]
+			groups[name] = append(groups[name], fl)
+		}
+	})
+
+	groupsOrdered := make(sort.StringSlice, 0, len(groups))
+	for groupName := range groups {
+		groupsOrdered = append(groupsOrdered, groupName)
+	}
+	sort.Sort(groupsOrdered)
+
+	fmt.Fprintf(os.Stderr, "Usage: %s [options ...] -config.file=<config_file>:\n\n", os.Args[0])
```

Comment:
Is it really necessary to treat `config.file` specially? It's not more required than other flags, if a config file named `prometheus.conf` is available, it doesn't need to be set explicitly.